### PR TITLE
reveals silos, if multi silo special is active

### PR DESCRIPTION
**Pull request by Arktischer**
to let people know how many silos are still alive, they are revealed (with fish)
### Tested Changes:
- [ x] I've tested the changes locally or with people.
- [ ] I've not tested the changes.

### CODE REVIEW

File: maps/biter_battles_v2/gui.lua [hunk 1]
```diff
@@ -1069,10 +1069,15 @@ function Public.spy_fish()
...
             local r = 96
             local surface = game.surfaces[global.bb_surface_name]
-            for _, player in pairs(game.forces[f[2]].connected_players) do
+            local entities = table.deep_copy(game.forces[f[2]].connected_players)
+            --reveal silos, if multi-silo special is active
+            if global.multi_silo then
+                table.add_all(entities, global.multi_silo[f[2]])
+            end
+            for _, entity in pairs(entities) do
                 game.forces[f[1]].chart(surface, {
```
Comment (clifffrey) [inline maps/biter_battles_v2/gui.lua:1078]: I'd be a bit more comfortable if this had something like `if entity.valid` - specifically the silos in multi-silo can die at almost any point, and I don't know whether this code will run before or after the `on_entity_died` event or whatever.

File: maps/biter_battles_v2/gui.lua [hunk 2]
```diff
@@ -1069,10 +1069,15 @@ function Public.spy_fish()
         if global.spy_fish_timeout[f[1]] - game.tick > 0 then
             local r = 96
             local surface = game.surfaces[global.bb_surface_name]
-            for _, player in pairs(game.forces[f[2]].connected_players) do
+            local entities = table.deep_copy(game.forces[f[2]].connected_players)
+            --reveal silos, if multi-silo special is active
+            if global.multi_silo then
+                table.add_all(entities, global.multi_silo[f[2]])
```
Comment (clifffrey) [inline maps/biter_battles_v2/gui.lua:1075]: maybe add both forces silos?  I could imagine wanting to know where the silos are on my own team as well...

Comment (clifffrey) [reply maps/biter_battles_v2/gui.lua:1075]: Relatedly (and less important), we could theoretically always reveal their locations to spectators in this function as well (i.e. unconditionally doing it for `game.forces.spectator.chart` during the special)

**Review by clifffrey — CHANGES_REQUESTED** [review]
I'm excited about this change! It will make the special better

File: maps/biter_battles_v2/gui.lua (hunk 2 above)
Comment (Arktischer) [reply maps/biter_battles_v2/gui.lua:1075]: if you add both forces silos you have somewhat unexpected behavior in that you have to send fish to reveal your own silo.. better i do that before the fish check

**Comment by Arktischer** [comment]
its getting late for me.. i think this should work, but it shows enemy silos without fish. Dont know why, probably just tired and its obvious. Will fix it tomorrow

**Comment by Arktischer** [comment]
game.forces.spectator.chart() reveals map, even if im in a team and
game.forces.spectator.connected_players is empty.
in spectator mode game.forces.north.chart() also reveals map, if i joined north before.
Do you know anything about that ?

**Comment by Arktischer** [comment]
.chart() for north or south always gets revealed for spectator.. i guess it makes sense this way. That means its not necessary to chart explicitly for spectators

**Comment by Arktischer** [comment]
should work now

**Comment by developer-8** [comment]
These changes should be made to the special game code and not to the scenario code.

**Comment by clifffrey** [comment]
We should actually add the multi-silo special to the scenario, it is run often enough that I don't want to break it when we next refactor something.

**Comment by Arktischer** [comment]
> These changes should be made to the special game code and not to the scenario code.

I do agree, but is that something i can help with? Where would i find the up to date multi-silo code ?

**Comment by developer-8** [comment]
> > These changes should be made to the special game code and not to the scenario code.
> 
> I do agree, but is that something i can help with? Where would i find the up to date multi-silo code ?

Idk if it up to date but here https://discord.com/channels/823696400797138974/1264598829559189565

**Comment by Arktischer** [comment]
I moved it to the multi-silo code, i will post the proposed changes in the mentioned dc channel.

**Pull request closed without merging**